plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt

```diff
@@ -245,10 +248,8 @@ private fun getPackageReferences(podfileLock: File): Set<PackageReference> {
             else -> node.textValue()
         }
 
-        val (name, version) = NAME_AND_VERSION_REGEX.find(entry)!!.groups.let {
-            it[1]!!.value to it[2]!!.value.removeSurrounding("(", ")")
-        }
-        versionForName[name] = version
+        val (name, version) = parseNameAndVersion(entry)
+        if (version != null) versionForName[name] = version
```

Comment:
Previously, for all names an entry existed in `versionForName`.
Now this does not hold anymore.

Can you explain why this is not problematic in line 261: `dependenciesForName.getValue(name)` ?

Reply:
> Now this does not hold anymore.

That's not actually true. The `version != null` condition will always be `true` here. Previously, this was hidden behind the `!!` assertions which I could have used here as well, but I consider their use as unclean. I'll change the code to use `checkNotNull()` instead to more clearly mark this as a precondition.